src/protocol/attribution/input.rs

```diff
@@ -19,9 +19,52 @@ use std::marker::PhantomData;
 use typenum::Unsigned;
 
 //
-// `accumulate_credit` protocol
+// 'apply_attribution_window` protocol
 //
 #[derive(Debug)]
+pub struct ApplyAttributionWindowInputRow<F: Field, BK: Fp2Array> {
+    pub timestamp: AdditiveShare<F>,
+    pub is_trigger_report: AdditiveShare<F>,
+    pub helper_bit: AdditiveShare<F>,
+    pub breakdown_key: XorShare<BK>,
+    pub trigger_value: AdditiveShare<F>,
+}
+
+#[derive(Debug)]
+pub struct MCApplyAttributionWindowInputRow<F: Field, T: Arithmetic<F>> {
+    pub timestamp: T,
+    pub is_trigger_report: T,
+    pub helper_bit: T,
+    pub breakdown_key: Vec<T>,
+    pub trigger_value: T,
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field, T: Arithmetic<F>> MCApplyAttributionWindowInputRow<F, T> {
+    pub fn new(
+        timestamp: T,
+        is_trigger_report: T,
+        helper_bit: T,
+        breakdown_key: Vec<T>,
+        trigger_value: T,
+    ) -> Self {
+        Self {
+            timestamp,
+            is_trigger_report,
+            helper_bit,
+            breakdown_key,
+            trigger_value,
+            _marker: PhantomData,
+        }
+    }
+}
+
+pub type MCApplyAttributionWindowOutputRow<F, T> = MCAccumulateCreditInputRow<F, T>;
```

Comment:
maybe we can work on more concise naming here. this type is in inside `attribution` module so we can eliminate the `attribution`word from it. In general it feels like the input and output for each circuit could be expressed as `protocol::circuit::input[row]`, `protocol::circuit::output[row]`. For this example, they could look like this

```rust
mod attribution {
     ... 
    mod window {
         ...
         pub struct InputRow<F, T> { ... }
         pub type OutputRow<F, T> = ...
    
         async fn apply<F, C, T>(ctx: C, input: &[InputRow<F, T>], ...
) -> Result<Vec<OutputRow<F, T>>, Error> { .. }
    }
}

```

Reply:
I do agree with your idea of separating words into module paths we discussed in NYC. That said, this particular case, "Attribution Window", the words are inseparable as it's a technical term. It's like separating `SemiHonestContext` to `mod semi { mod honest { struct Context } }`.